homeassistant/components/withings/__init__.py

```diff
@@ -101,35 +102,39 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     return True
 
 
-async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
-    """Set up Withings from a config entry."""
-    config_updates: dict[str, Any] = {}
-
-    # Add a unique id if it's an older config entry.
-    if entry.unique_id != entry.data["token"]["userid"] or not isinstance(
-        entry.unique_id, str
-    ):
-        config_updates["unique_id"] = str(entry.data["token"]["userid"])
-
-    # Add the webhook configuration.
-    if CONF_WEBHOOK_ID not in entry.data:
-        webhook_id = webhook.async_generate_id()
-        config_updates["data"] = {
-            **entry.data,
-            **{
-                const.CONF_USE_WEBHOOK: hass.data[DOMAIN][const.CONFIG][
-                    const.CONF_USE_WEBHOOK
-                ],
-                CONF_WEBHOOK_ID: webhook_id,
-            },
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
```

Comment:
I don't see the need to migrate. We prefer doing soft migrations over hard migrations, to allow roll backs.

We don't need to remove the profile from the config entry data of old config entries. Just stop using that item.